public/app/features/variables/pickers/OptionsPicker/reducer.ts

```diff
@@ -178,20 +187,39 @@ const optionsPickerSlice = createSlice({
       };
     },
     toggleAllOptions: (state, action: PayloadAction): OptionsPickerState => {
-      if (state.selectedValues.length > 0) {
+      // Check if 'All' option is configured and if it's selected
+      const isAllSelected = state.selectedValues.find((option) => option.value === ALL_VARIABLE_VALUE);
+      const isAllConfigured = state.options.find((option) => option.value === ALL_VARIABLE_VALUE);
+
+      // If 'All' option is not selected but some options are, clear all options and select 'All'
+      if (state.selectedValues.length > 0 && isAllConfigured && !isAllSelected) {
         state.selectedValues = [];
+        const allOptionState = state.options.find((option) => option.value === ALL_VARIABLE_VALUE);
```

Comment:
you already have this object in `isAllConfigured`. I would reuse it, instead of iterating again over the list (maybe with the name of `allOptionState`)

Reply:
and I would change the if as 
if (state.selectedValues.length > 0 && !!allOptionState && !isAllSelected)

(the name of the object and the !! to make it a boolean value)